nntrainer/layers/layer_node.cpp

```diff
@@ -104,16 +104,19 @@ LayerNode::LayerNode(std::unique_ptr<nntrainer::Layer> &&l, size_t idx) :
 
 int LayerNode::setProperty(std::vector<std::string> properties) {
   int status = ML_ERROR_NONE;
+  bool already_distributed =
+    !std::get<props::Distribute>(*layer_node_props).empty() &&
+    std::get<props::Distribute>(*layer_node_props).get();
```

Comment:
I think those are same. if it is empty, get() return false..

Reply:
if the property is empty(), get() throws. Hence, the check of empty() before get().